src/main/webapp/WEB-INF/view/conversations.jsp

```diff
@@ -61,7 +63,14 @@
       for(Conversation conversation : conversations){
     %>
       <li><a href="/chat/<%= conversation.getTitle() %>">
-        <%= conversation.getTitle() %></a></li>
+        <%= conversation.getTitle() %></a>
+        <% String u = (String) request.getSession().getAttribute("user");
+        if(u != null){
+            boolean check = (boolean) request.getSession().getAttribute("isSuperUser");
+            if(check){ %>
+            <a href=""onclick="deleteConversation(<%=conversation.getId()%>)">delete</a>
```

Comment:
Also, "deleteConversation...;" will not work since you don't have access to the ConversationStore to call deleteConversation. You can either give the conversation.jsp the conversation store as an attribute or add a Delete request to the ConversationServlet that calls delete in the ConversationStore (this is probably better).

Reply:
I hadn't coded the deleteConversation function yet. I will use that to send a request to doDelete in the conversation servlet. Thanks though!